docs/enhancements/0001_disconnected.md

```diff
@@ -0,0 +1,93 @@
+# Disconnected support for KMM
+
+**Goal**: describe the requirements and implementation ideas for KMM disconnected support
+
+## Introduction: disconnected in OCP
+
+The term "disconnected clusters" refers to clusters that do not have Internet access and thus cannot download publicly
+available container images.
+To support disconnected clusters, a private registry must be used.
+The registry is installed and managed by the cluster administrator.
+
+Red Hat offers [documentation](https://docs.openshift.com/container-platform/4.12/installing/disconnected_install/installing-mirroring-installation-images.html#installing-mirroring-installation-images)
+to mirror the OpenShift system iamges on a private registry.
+Under the hood, the tools used:
+- copy the container images from the public registry on the Internet to the private one;
+- create [`ImageContentSourcePolicy`] objects that configure the mirroring on nodes.
+  [`ImageContentSourcePolicy`] objects configure alternate locations for images specified **by digest only**.
+
+KMM itself can be mirrored like any operator with the usual tools.  
+However, some use cases require the controllers and other components to pull and push images from and to registries.
+The operator does not inherit the mirroring configuration of the container runtime on the node, so by default it will
+not pull images from mirrored registries.
+In those cases, support for mirrored registries as configured in [`ImageContentSourcePolicy`] objects must be
+implemented in the operator code.
+
+## In KMM
+
+The scenarios where the operator or one of its components might need to build an image are listed below:
+
+| Scenario                                                                            | Handled by OCP |  Work needed  |
+|-------------------------------------------------------------------------------------|:--------------:|:-------------:|
+| When the `Module` reconciler checks if the output image already exists              |       ❌        | ❌<sup>1</sup> |
+| When the `PreflightValidation` reconciler checks if the output image already exists |       ❌        | ❌<sup>1</sup> |
+| When pulling the base image (`FROM` in the `Dockerfile`)                            | ✅<sup>2</sup>  |       ❌       |
```

Comment:
Maybe worth mentioning that the registry auth secret will need to fit the mirrored registry and not the public one.